include/alpaka/mem/buf/BufUniformCudaHipRt.hpp

```diff
@@ -32,29 +32,44 @@ namespace alpaka
     template<typename TElem, typename TDim, typename TIdx>
     class BufCpu;
 
+    namespace detail
+    {
+        template<typename TDim, typename SFINAE = void>
+        struct PitchHolder
+        {
+            explicit PitchHolder(std::size_t)
+            {
+            }
+        };
+
+        template<typename TDim>
+        struct PitchHolder<TDim, std::enable_if_t<TDim::value >= 2>>
+        {
+            std::size_t m_rowPitchInBytes;
+        };
+    } // namespace detail
+
     //! The CUDA/HIP memory buffer.
     template<typename TApi, typename TElem, typename TDim, typename TIdx>
-    class BufUniformCudaHipRt : public internal::ViewAccessOps<BufUniformCudaHipRt<TApi, TElem, TDim, TIdx>>
+    struct BufUniformCudaHipRt
+        : detail::PitchHolder<TDim>
+        , internal::ViewAccessOps<BufUniformCudaHipRt<TApi, TElem, TDim, TIdx>>
     {
-    public:
-        static_assert(
-            !std::is_const_v<TElem>,
-            "The elem type of the buffer can not be const because the C++ Standard forbids containers of const "
-            "elements!");
-        static_assert(!std::is_const_v<TIdx>, "The idx type of the buffer can not be const!");
+        static_assert(!std::is_const_v<TElem>, "The elem type of the buffer must not be const");
+        static_assert(!std::is_const_v<TIdx>, "The idx type of the buffer must not be const!");
 
         //! Constructor
         template<typename TExtent, typename Deleter>
         ALPAKA_FN_HOST BufUniformCudaHipRt(
             DevUniformCudaHipRt<TApi> const& dev,
             TElem* const pMem,
             Deleter deleter,
-            std::size_t pitchBytes,
-            TExtent const& extent)
-            : m_dev(dev)
-            , m_extentElements(getExtentVecEnd<TDim>(extent))
+            TExtent const& extent,
+            std::size_t pitchBytes)
```

Comment:
why the swap in the order of the arguments ?

Reply:
Because I initially thought about having a ctor overload without the `pitch`, but then I decided against it. I can order them back.

Reply:
I restored the original order.